synapse/server.py

```diff
@@ -388,7 +388,9 @@ def get_federation_http_client(self) -> MatrixFederationHttpClient:
         tls_client_options_factory = context_factory.FederationPolicyForHTTPS(
             self.config
         )
-        return MatrixFederationHttpClient(self, tls_client_options_factory)
+        return MatrixFederationHttpClient(
+            self, tls_client_options_factory, https_proxy=os.getenvb(b"HTTPS_PROXY"),
```

Comment:
I suspect we would also want to support `http_proxy` (which I suspect is not too much more work).

Reply:
Is it possible that federation requests could end up with a http target? I was under the impression that https is a requirement for federation?

Reply:
The `HTTP_PROXY` and `HTTPS_PROXY` variables refer to the type of connection between the client and the proxy, not the kind of requests that the proxy will be proxying upstream. The proxy should support both HTTP and HTTPS requests regardless of this setting.

Reply:
Though actually, I always feel a bit insane when talking about this, because it seems there are programs that flat out disagree on this. I just noticed your #9090 🤔

Reply:
Arg I seem to have made the same mistake as @dkasak in getting those two confused. I think the federation APIs only talk via HTTPS.